cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -1266,6 +1266,221 @@ pub(crate) fn lower_load<C: LowerCtx<I = Inst>, F: FnMut(&mut C, Writable<Reg>,
     f(ctx, rd, elem_ty, mem);
 }
 
+pub(crate) fn emit_shl_i128<C: LowerCtx<I = Inst>>(
+    ctx: &mut C,
+    src: ValueRegs<Reg>,
+    dst: ValueRegs<Writable<Reg>>,
+    amt_src: Reg,
+) {
+    let src_lo = src.regs()[0];
+    let src_hi = src.regs()[1];
+    let dst_lo = dst.regs()[0];
+    let dst_hi = dst.regs()[1];
+
+    //   and  amt, amt, #127
```

Comment:
You are right that this instruction sequence can be improved - see what Clang is generating in the same situation [here](https://godbolt.org/z/edKP1c6K6).

A similar improvement is possible for the right shifts.

Reply:
Oops, i should have checked that.

I've fixed the code so that it now emits a similar sequence to what clang is doing.